fuel-vm/src/interpreter/memory.rs

```diff
@@ -707,8 +712,9 @@ mod tests {
         => (false, [0u8; 100]); "Internal too large for heap"
     )]
     fn test_mem_write(addr: usize, data: &[u8], registers: OwnershipRegisters) -> (bool, [u8; 100]) {
-        let mut memory = [0u8; 100];
-        let r = try_mem_write(addr, data, registers, &mut memory[..]).is_ok();
+        let mut memory: Box<[u8; VM_MEMORY_SIZE]> = vec![0u8; VM_MEMORY_SIZE].try_into().unwrap();
```

Comment:
[nit] wouldn't it be cleaner to just use `Box::new` here instead of `try_into().unwrap()`?

```suggestion
        let mut memory: Box<[u8; VM_MEMORY_SIZE]> = Box::new(vec![0u8; VM_MEMORY_SIZE]);
```